common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -1069,15 +1104,19 @@ private void logSecureHardwareState(@NonNull final KeyPair kp) {
                 );
                 final KeyInfo info = factory.getKeySpec(privateKey, KeyInfo.class);
                 final boolean isInsideSecureHardware = info.isInsideSecureHardware();
-                msg = "SecretKey is secure hardware backed? " + isInsideSecureHardware;
+                Logger.info(TAG, "SecretKey is secure hardware backed? " + isInsideSecureHardware);
+                return isInsideSecureHardware
+                        ? SecureHardwareState.TRUE_UNATTESTED
+                        : SecureHardwareState.FALSE;
             } catch (final Exception e) {
-                msg = "Failed to query secure hardware state.";
+                Logger.info(TAG, "Failed to query secure hardware state.");
+                return SecureHardwareState.UNKNOWN_QUERY_ERROR;
```

Comment:
This is a really broad catch.  Can we narrow this at all, or, if we cannot, can we at least log some exception information here so that we can eventually narrow this down?  And if we have the ability to stuff back some telemetry, the class of the exception that's thrown here might be very interesting.

Reply:
Added better logging, reduced the scope of the catch block
8ae660a5e690dfbc092331ff2ab77668ff568ad0